### PR TITLE
feat: add unlabeled data classes

### PR DESCRIPTION
**Pull request by tadejsv**
TODO:

- [x] update documentation

### CODE REVIEW

File: docs/basics/data-format.md [hunk 1]
```diff
@@ -17,13 +17,14 @@ finetuner.fit(model,
...
-There are three different types of datasets:
+There are four different types of datasets:
 
 ```{toctree}
 :maxdepth: 1
 
 datasets/class-dataset
 datasets/session-dataset
+datasets/instance-dataset
 datasets/unlabeled-dataset
```
Comment (bwanglzu) [inline docs/basics/data-format.md:28]: where is this unlabeled dataset comes from?

File: finetuner/tuner/dataset/samplers.py [hunk 2]
```diff
@@ -7,6 +7,46 @@
...
+
+        :param num_instances: Number of instances in the dataset
+        :param batch_size: How many items to include in a batch
+        :param repeat_instance: How many times to repeat each instance in a batch
+        """
+
+        if batch_size <= 0:
+            raise ValueError('batch_size must be a positive integer')
+
+        if batch_size % repeat_instance != 0:
```
Comment (bwanglzu) [inline finetuner/tuner/dataset/samplers.py:28]: we might also don't want to set `repeat_instance` to 1? otherwise SSL learns nothing

**Review by bwanglzu — CHANGES_REQUESTED** [review]
minor comments

File: docs/basics/data-format.md (hunk 1 above)
Comment (tadejsv) [reply docs/basics/data-format.md:28]: It's already there (check the documentation). It's basically only used in Labeler now

File: finetuner/tuner/dataset/samplers.py (hunk 2 above)
Comment (tadejsv) [reply finetuner/tuner/dataset/samplers.py:28]: Good idea

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Pull request merged**